Feature/onboarding 1.1 ~ 1.6

ihor-stack opened the pull request.

CrazyOptimist commented on src/features/onboarding/register/StepPayment.tsx:
```diff
@@ -0,0 +1,116 @@
+        break
+
+      case PaymentMethods.PslAddress:
+        props.setExchangeAddress('lkj3432hghg41hg32hg1h3g2h1g3g21')
```
This will be changed later anyway, so I guess okay for now.

CrazyOptimist commented on src/features/onboarding/welcome/Welcome.tsx:
```diff
@@ -1,42 +1,56 @@
-import { colors } from '../../../common/theme/colors'
 import * as ROUTES from '../../../common/utils/constants/routes'
-import * as Styles from './Welcome.styles'
+import styles from './Welcome.css'
```
Please rename Welcome.css to Welcome.module.css

ihor-stack replied:
renamed

CrazyOptimist commented on src/legacy/Routes.tsx:
```diff
@@ -499,7 +500,8 @@ class RouteApp extends React.Component<any, any> {
         <GlitchImageModal />
-        {info?.version && <Header />}
+        {/* info?.version && <Header /> */}
+        <Header />
```
What was fixed by this change? Code in legacy is unpredictable, so I would recommend to not change it without necessity

ihor-stack replied:
When user passed login/register steps and redirected to dashboard or another page (legacy routing), header is hidden without this fix.

ngvtuan commented on src/features/onboarding/newPassword/NewPassword.tsx:
```diff
@@ -3,10 +3,11 @@ import { Link } from 'react-router-dom'
+import PasswordStrength from 'common/components/PasswordStrength/PasswordStrength'
 import { colors } from '../../../common/theme/colors'
 import * as ROUTES from '../../../common/utils/constants/routes'
+//import { calcPasswordStrength } from 'common/utils/passwords'
```
Remove the un-use code

ihor-stack replied:
uncommented & used

ngvtuan commented on tailwind.config.js:
```diff
@@ -368,6 +371,7 @@ module.exports = {
         '598px': '37.375rem',
         '690px': '43.125rem',
         '700px': '43.75rem',
+        '95p': '95%',
```
I think the name should be `95/100`

ihor-stack replied:
removed (not used)

ngvtuan reviewed (commented):
![ui issues 1](https://user-images.githubusercontent.com/10944387/123766649-11a9f100-d8f1-11eb-9105-0296b28ed502.jpg)
![ui issues 2](https://user-images.githubusercontent.com/10944387/123766694-1bcbef80-d8f1-11eb-82e0-b2114ae1540c.jpg)
<img width="541" alt="Text should be Primary login" src="https://user-images.githubusercontent.com/10944387/123766878-428a2600-d8f1-11eb-84ad-df8b87bcb981.png">
![ui issues 5](https://user-images.githubusercontent.com/10944387/123766991-5897e680-d8f1-11eb-965d-3e57a3cc1e86.jpg)
![ui issues 6](https://user-images.githubusercontent.com/10944387/123766999-59c91380-d8f1-11eb-8640-040d40bf0f11.jpg)
![ui issues 7](https://user-images.githubusercontent.com/10944387/123767001-5a61aa00-d8f1-11eb-936f-1789330492ba.jpg)
![ui issues 8](https://user-images.githubusercontent.com/10944387/123767002-5afa4080-d8f1-11eb-9b81-313ed7512045.jpg)
![ui issues 9](https://user-images.githubusercontent.com/10944387/123767004-5b92d700-d8f1-11eb-88c0-9eab73fffecf.jpg)

![ui issues 3](https://user-images.githubusercontent.com/10944387/123766922-4a49ca80-d8f1-11eb-98aa-ef0c60d740a2.jpg)
<img width="1097" alt="remove tooltip word in the tooltip popup" src="https://user-images.githubusercontent.com/10944387/123766938-4d44bb00-d8f1-11eb-9fa6-6522b7eaa638.png">
![ui issues 4](https://user-images.githubusercontent.com/10944387/123766963-53d33280-d8f1-11eb-9607-fa6438315fb7.jpg)
Please skip `1` because i think it is the design issue.

ihor-stack commented on src/common/assets/icons/ico-times-old.svg:
```diff
@@ -0,0 +1,3 @@
+<svg width="12" height="12" viewBox="0 0 12 12" fill="none" xmlns="http://www.w3.org/2000/svg">
```
removed

ihor-stack commented on src/common/components/PasswordStrength/PasswordStrength.tsx:
```diff
@@ -0,0 +1,56 @@
+}
+
+const PasswordStrength = (props: TPasswordStrengthProps): JSX.Element => {
+  let cl = ''
```
changed

ihor-stack commented on src/common/components/PasswordStrength/PasswordStrength.tsx:
```diff
@@ -0,0 +1,56 @@
+        props.className ? props.className : '',
+      )}
+    >
+      {items.map((e, i) => (
```
i is index here
"e" changed

ihor-stack commented on src/common/components/Tooltip/Tooltip.tsx:
```diff
@@ -16,16 +17,17 @@ const Tooltip: FunctionComponent<TTooltipProps> = ({
   content,
   width,
   children,
+  hPosPercent = 150,
```
Sorry, vPos should be there. Fixed.
In some cases tooltip shows too high with hardcoded (earlier) value 150%.

ihor-stack commented on src/common/utils/passwords.ts:
```diff
@@ -0,0 +1,48 @@
+  type Obj1 = {
+    [key: string]: number
+  }
+  const letters: Obj1 = {}
```
replaced

ihor-stack commented on src/common/utils/passwords.ts:
```diff
@@ -0,0 +1,48 @@
+  type Obj2 = {
+    [key: string]: boolean
+  }
+  const variations: Obj2 = {
```
replaced

ihor-stack commented on src/features/onboarding/newPassword/NewPassword.tsx:
```diff
@@ -33,6 +34,9 @@ const NewPassword: React.FC = () => {
   }
 
+  // calcPasswordStrength() may be used to calc strength
+  const pwdStrength = 0
```
used this function

ihor-stack commented on src/features/onboarding/register/Register.tsx:
```diff
@@ -1,109 +1,224 @@
+      state.paymentMethod,
+      state.promoCode,
+      state.exchangeAddress,
+    */
```
it's place to save new user data. Comment about it

ihor-stack commented on src/features/onboarding/register/StepBackup.tsx:
```diff
@@ -0,0 +1,142 @@
+    },
+  ]
+
+  const onDlPdf = () => {
```
renamed to onDownloadPdf

ihor-stack commented on src/features/onboarding/register/StepBackup.tsx:
```diff
@@ -0,0 +1,142 @@
+
+              {pdfPrepareProgress === 0 && (
+                <div className='flex-grow flex justify-end'>
+                  <span
```
changed

ihor-stack commented on src/features/onboarding/register/StepFee.tsx:
```diff
@@ -0,0 +1,172 @@
+
+  const toClipboard = () => {
+    setCopiyng(true) // animate icon
+    setTimeout(() => setCopiyng(false), 200)
```
fixed

ihor-stack commented on src/common/components/Inputs/Input.css:
```diff
@@ -0,0 +1,20 @@
+.hintAsTooltip {
+  display: none;
```
It will be better to leave display prop here. CSS used to show tooltip when input has focus.

ihor-stack commented on src/common/routes/Routes.tsx:
```diff
@@ -1,7 +1,7 @@
 import { RouteComponentProps } from 'react-router'
 import { Route, Switch, useLocation, useHistory } from 'react-router-dom'
-import { CSSTransition, TransitionGroup } from 'react-transition-group'
+// import { CSSTransition, TransitionGroup } from 'react-transition-group'
```
removed

ihor-stack commented on src/common/routes/Routes.tsx:
```diff
@@ -57,13 +57,18 @@ const Routes: React.FC<RoutesProps> = ({ setUser }) => {
           </Switch>
         </CSSTransition>
       </TransitionGroup>
+      */}
```
Not sure if this is a bug or something else. Removed

ihor-stack commented on src/features/onboarding/register/Regiser.state.ts:
```diff
@@ -0,0 +1,117 @@
+import { useState } from 'react'
+// import * as ROUTES from 'common/utils/constants/routes'
```
removed

ihor-stack commented on src/features/onboarding/register/Buttons.tsx:
```diff
@@ -0,0 +1,45 @@
+  return (
+    <button
+      onClick={() => {
+        if (props.active) {
```
ok, modified

ihor-stack commented on src/features/onboarding/register/Register.css:
```diff
@@ -0,0 +1,17 @@
+}
+
+.qrWrapper {
+  box-shadow: 0px 64px 64px -48px rgba(31, 47, 70, 0.12);
```
changed

ihor-stack commented on src/features/onboarding/register/StepPayment.tsx:
```diff
@@ -0,0 +1,116 @@
+          1,000 PSL fee (worth ~$5.20 at current PSL price)
+        </h2>
+        <div className='mt-7'>
+          {methods.map((m, i) => (
```
changed m -> method
i is index

ihor-stack commented on src/features/onboarding/register/StepRegister.tsx:
```diff
@@ -0,0 +1,126 @@
+          hintAsTooltip={true}
+        />
+        <div className='mt-6'>
+          <Input
```
changed to InputPassword

ihor-stack commented on src/features/onboarding/welcome/Welcome.css:
```diff
@@ -0,0 +1,8 @@
+.playBtn {
```
yes, if add to tailwind config. But these styles are used here only.

ihor-stack commented on src/features/onboarding/register/Buttons.tsx:
```diff
@@ -0,0 +1,45 @@
+  onClick(): void
+}
+
+export const NextButton = (props: TNextBtnProps): JSX.Element => {
```
because common Button have no disabled state/props

ngvtuan reviewed (commented):
![ui issue 1](https://user-images.githubusercontent.com/10944387/124245692-6b592800-db4a-11eb-823a-7fff7989d8e4.jpg)
![ui issue 2](https://user-images.githubusercontent.com/10944387/124245714-714f0900-db4a-11eb-9f9b-bfcecc243642.jpg)
![ui issue 3](https://user-images.githubusercontent.com/10944387/124245713-714f0900-db4a-11eb-9851-696d05a5deba.jpg)

ihor-stack commented:
Fixed.
Problem with download icon was mask in svg. Looks like browsers and Electron's Chromium engine display it a little bit different.
Removed mask, now icon looks ok.

ngvtuan reviewed (commented):
<img width="1339" alt="ui issue" src="https://user-images.githubusercontent.com/10944387/124342441-343c5280-dbee-11eb-8633-ee36b32a02ac.png">
<img width="1440" alt="warning" src="https://user-images.githubusercontent.com/10944387/124342444-3a323380-dbee-11eb-86a3-cd45af56a657.png">
